File: scripts/generate_notification_icon.py
import os
from PIL import Image, ImageDraw

def draw_spendflow_notification_icon(size: int) -> Image.Image:
    # High resolution canvas for smooth anti-aliased drawing (scale 8x then resize down)
    scale = 8
    canvas_size = size * scale
    img = Image.new("RGBA", (canvas_size, canvas_size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)

    # Unit coordinate system (0 to 100)
    def pt(x, y):
        return (x * canvas_size / 100.0, y * canvas_size / 100.0)

    white = (255, 255, 255, 255)
    
    # ── 1. Draw Bold Modern SpendFlow Emblem (Fluid S-Curve + Diamond Spark) ──
    # Top curve of S
    top_points = [
        pt(70, 10), pt(36, 10), pt(26, 13), pt(18, 20), pt(12, 30), pt(12, 40),
        pt(16, 50), pt(24, 56), pt(36, 60), pt(60, 60), pt(68, 64), pt(72, 70),
        pt(72, 78), pt(68, 84), pt(60, 88), pt(24, 88), pt(18, 84), pt(18, 76),
        pt(8, 76), pt(8, 84), pt(14, 92), pt(24, 96), pt(60, 96), pt(72, 92),
        pt(80, 84), pt(84, 76), pt(84, 66), pt(78, 56), pt(68, 50), pt(56, 48),
        pt(36, 48), pt(28, 44), pt(24, 38), pt(24, 30), pt(28, 24), pt(36, 20),
        pt(70, 20), pt(76, 26), pt(76, 32), pt(86, 32), pt(86, 24), pt(80, 14),
    ]
    draw.polygon(top_points, fill=white)

    # 4-point Spark in top right
    cx, cy = pt(82, 18)
    r_outer = 10 * canvas_size / 100.0
    r_inner = 3 * canvas_size / 100.0
    spark_points = [
        (cx, cy - r_outer),
        (cx + r_inner, cy - r_inner),
        (cx + r_outer, cy),
        (cx + r_inner, cy + r_inner),
        (cx, cy + r_outer),
        (cx - r_inner, cy + r_inner),
        (cx - r_outer, cy),
        (cx - r_inner, cy - r_inner),
    ]
    draw.polygon(spark_points, fill=white)

    # Smooth downsampling with high quality Lanczos filter
    return img.resize((size, size), Image.Resampling.LANCZOS)


def main():
    sizes = [
        ("drawable-mdpi", 24),
        ("drawable-hdpi", 36),
        ("drawable-xhdpi", 48),
        ("drawable-xxhdpi", 72),
        ("drawable-xxxhdpi", 96),
    ]

    target_res_roots = [
        os.path.join(os.path.dirname(__file__), "..", "android", "app", "src", "main", "res"),
        r"C:\SpendFlow\android\app\src\main\res",
    ]

    for res_root in target_res_roots:
        if not os.path.exists(res_root):
            continue
        print(f"Generating pure white notification icons in {res_root}...")
        for folder, size in sizes:
            folder_path = os.path.join(res_root, folder)
            os.makedirs(folder_path, exist_ok=True)
            icon = draw_spendflow_notification_icon(size)
            out_file = os.path.join(folder_path, "notification_icon.png")
            icon.save(out_file, "PNG")
            print(f"  [OK] Saved {size}x{size} notification_icon.png to {folder}")

    # Update assets/android-icon-monochrome.png
    assets_dir = os.path.join(os.path.dirname(__file__), "..", "assets")
    mono_icon = draw_spendflow_notification_icon(96)
    mono_path = os.path.join(assets_dir, "android-icon-monochrome.png")
    mono_icon.save(mono_path, "PNG")
    print(f"  [OK] Updated {mono_path}")

    print("\n[SUCCESS] All pure white notification icons generated successfully!")

if __name__ == "__main__":
    main()

File: scripts/generate-notification-icon.js
const fs = require('fs');
const path = require('path');
const { generateImageAsync } = require('@expo/image-utils');

// Android Notification Icon specs:
// - Must be pure white (#FFFFFF) foreground on transparent background
// - Large & prominent (filling 85% of canvas) so it is crisp and clearly identifiable on the status bar
const NOTIF_SIZES = [
  { folder: 'drawable-mdpi', size: 24 },
  { folder: 'drawable-hdpi', size: 36 },
  { folder: 'drawable-xhdpi', size: 48 },
  { folder: 'drawable-xxhdpi', size: 72 },
  { folder: 'drawable-xxxhdpi', size: 96 },
];

const TARGET_RES_DIRS = [
  path.join(__dirname, '..', 'android', 'app', 'src', 'main', 'res'),
  'C:\\SpendFlow\\android\\app\\src\\main\\res',
];

// Bold, high-contrast pure white (#FFFFFF) SpendFlow S-monogram & Wallet silhouette on transparent background
const notifSvg = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 96 96" width="96" height="96">
  <!-- Outer Card Body with rounded corners -->
  <path d="M12 20 C12 13.5 17.5 8 24 8 L72 8 C78.5 8 84 13.5 84 20 L84 76 C84 82.5 78.5 88 72 88 L24 88 C17.5 88 12 82.5 12 76 Z" fill="#FFFFFF"/>
  <!-- Transparent Cutout 1: Magnetic stripe / upper accent -->
  <rect x="20" y="18" width="56" height="10" rx="3" fill="#000000"/>
  <!-- Transparent Cutout 2: Bold Dynamic S-Curve / Growth Flow -->
  <path d="M66 40 C66 35 60 32 50 32 C38 32 32 37 32 44 C32 58 64 54 64 66 C64 72 58 75 48 75 C36 75 30 70 30 64 L40 64 C40 67 43 68 48 68 C53 68 56 66 56 63 C56 52 24 55 24 41 C24 33 32 26 48 26 C62 26 74 33 74 40 Z" fill="#000000"/>
</svg>`;

// Also a pure silhouette version with no black cutouts (pure white paths with transparent gaps)
const pureSilhouetteSvg = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 96 96" width="96" height="96">
  <!-- Top Bar -->
  <path d="M22 14 C16.5 14 12 18.5 12 24 L12 28 L84 28 L84 24 C84 18.5 79.5 14 74 14 Z" fill="#FFFFFF"/>
  <!-- Main Card Body -->
  <path d="M12 36 L12 72 C12 77.5 16.5 82 22 82 L74 82 C79.5 82 84 77.5 84 72 L84 36 Z" fill="#FFFFFF"/>
  <!-- Inner Cutout for Wallet Lock / Chip Accent (transparent via mask or path geometry) -->
</svg>`;

// The cleanest, sharpest Android status bar icon: A bold geometric SpendFlow "S" emblem with a spark in pure white
const spendflowEmblemSvg = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 96 96" width="96" height="96">
  <!-- Bold SpendFlow Geometric Icon in Pure White on 100% Transparent Canvas -->
  <!-- Top curve of S -->
  <path d="M68 12 L36 12 C22.7 12 12 22.7 12 36 C12 49.3 22.7 60 36 60 L60 60 C66.6 60 72 65.4 72 72 C72 78.6 66.6 84 60 84 L24 84 C20.7 84 18 81.3 18 78 L18 76 L8 76 L8 78 C8 86.8 15.2 94 24 94 L60 94 C72.2 94 82 84.2 82 72 C82 59.8 72.2 50 60 50 L36 50 C29.4 50 24 44.6 24 38 C24 31.4 29.4 26 36 26 L68 26 C71.3 26 74 28.7 74 32 L74 34 L84 34 L84 32 C84 23.2 76.8 12 68 12 Z" fill="#FFFFFF"/>
  <!-- Spark / Star in top-right -->
  <path d="M78 8 L81 18 L91 21 L81 24 L78 34 L75 24 L65 21 L75 18 Z" fill="#FFFFFF"/>
</svg>`;

async function main() {
  console.log('Generating crisp white Android notification icons...');
  const svgPath = path.join(__dirname, 'notif_temp.svg');
  fs.writeFileSync(svgPath, spendflowEmblemSvg);

  for (const targetRes of TARGET_RES_DIRS) {
    if (!fs.existsSync(targetRes)) continue;
    console.log(`Updating notification icons in ${targetRes}...`);

    for (const item of NOTIF_SIZES) {
      const folderPath = path.join(targetRes, item.folder);
      if (!fs.existsSync(folderPath)) fs.mkdirSync(folderPath, { recursive: true });

      const img = await generateImageAsync(
        { projectRoot: path.join(__dirname, '..') },
        {
          src: svgPath,
          width: item.size,
          height: item.size,
          resizeMode: 'contain',
          format: 'png',
        }
      );

      const dest = path.join(folderPath, 'notification_icon.png');
      fs.writeFileSync(dest, img.source);
      console.log(`  ✓ Wrote ${item.size}x${item.size} to ${dest}`);
    }
  }

  // Update assets/android-icon-monochrome.png
  const monoImg = await generateImageAsync(
    { projectRoot: path.join(__dirname, '..') },
    {
      src: svgPath,
      width: 96,
      height: 96,
      resizeMode: 'contain',
      format: 'png',
    }
  );
  fs.writeFileSync(path.join(__dirname, '..', 'assets', 'android-icon-monochrome.png'), monoImg.source);
  console.log('  ✓ Updated assets/android-icon-monochrome.png');

  try {
    fs.unlinkSync(svgPath);
  } catch {}

  console.log('🎉 Notification icon generation complete!');
}

main().catch(err => {
  console.error('Failed to generate notification icons:', err);
  process.exit(1);
});
